docs/current/guides/snippets/github-google-cloud/github-nodejs.yml

```diff
@@ -35,6 +35,9 @@ jobs:
       -
         name: Install
         run: npm install
+      -
+        name: Install Dagger CLI
+        run: cd /usr/local && curl -L https://dl.dagger.io/dagger/install.sh | sh
       -
         name: Release and deploy with Dagger
-        run: node ci/index.mjs
+        run: dagger run node ci/index.mjs
```

Comment:
I would like this reviewed by others. 

As per the discussion in the source issue, the goal is to transition to "dagger run" in all cases. I've added it here in CI with this goal in mind and for consistency with the local pipeline run shown in the guide. However, it's worth noting that when running in CI, there is an extra step (downloading the Dagger CLI) and the live progress output is also disabled. Is it still correct to use "dagger run" in CI? 

WDYT @gerhard @shykes @vito ?

Reply:
It makes sense in my opinion, specially with Zenith, you'll simply execute `dagger do <task>` so you'll need the Dagger CLI.
Basically it's quite similar to what we're doing in our own ci

Reply:
There are a few things making me hesitate on approving this PR:

1. We're probably about to leapfrog `dagger run` with Zenith so I'm not sure it's worth rocking the boat in all the docs just yet
2. It seems like this will all just download the latest CLI which might not be compatible with the SDK version your code is using. But at the same time it will be kind of crummy to maintain a version number in two places (GHA YAML + your codebase). Feels like we need to figure out the bootstrapping story here a bit more. :thinking: (But I may be missing something.)
3. There is no material benefit to using `dagger run` in CI, since I think it'll just end up with the exact same console output format that you already see without it, you just had to do an extra step to download + install + align the version of the CLI yourself, instead of just letting the SDK do it

Reply:
I understand the concerns, but I think we should move forward with the change anyway.

> 1. We're probably about to leapfrog `dagger run` with Zenith so I'm not sure it's worth rocking the boat in all the docs just yet

That's true, but I think the benefits of starting to shift our UX towards CLI-first, outweigh the drawbacks of changing *how* to use the CLI again in another few months. Getting everyone to think of Dagger as a CLI tool first, instead of a library, is a big lift and it will take a while to get the message across. So I think we should spare no effort to get that started.

> 2. It seems like this will all just download the latest CLI which might not be compatible with the SDK version your code is using. But at the same time it will be kind of crummy to maintain a version number in two places (GHA YAML + your codebase). Feels like we need to figure out the bootstrapping story here a bit more. 🤔 (But I may be missing something.)

Yeah, we don't have this part fully figured out yet, but we do know we want the CLI to become the source of truth for versioning. So it's a step in the right direction I think, to get everyone to think of their Dagger version as the CLI version first and foremost. That *does* put more pressure on SDKs versioning to somehow fit into that model. But Zenith will provide the last part of that puzzle, by making it possible to manage SDK versioning from the CLI. I feel like we'll be able to land on our feet here.

> 3. There is no material benefit to using `dagger run` in CI, since I think it'll just end up with the exact same console output format that you already see without it, you just had to do an extra step to download + install + align the version of the CLI yourself, instead of just letting the SDK do it

For TUI output that may be true (although CI tools sometimes give you a tty to watch in real time, don't they?). But I think there are still benefits:

* Starts the process of training CI administrators to think "CLI-first". The sooner we start the process, the sooner we'll have good docs, useful feedback from admins, etc
* Consistency is good. I'd rather have `dagger run` everywhere, even where it doesn't have much benefit, than a "well it depends" situation where users (and docs) have to carry the weight of deciding which mode to use where.
* I think the CLI may actually have tangible benefits in CI soon, in particular around 1) custom engine drivers, and 2) Dagger Cloud integration. Possibly also 3) secrets?

Reply:
Sounds good!

> although CI tools sometimes give you a tty to watch in real time, don't they

The only one I know of is [Concourse](https://github.com/concourse/concourse/blob/3793327b7c279f6e9d7638f8d410444e0b986a9e/atc/exec/task_step.go#L298-L303), since we really wanted maximum prettiness in log output. Every other CI/CD system I've come across sends stdout to a regular file or pipe. Now I'm really curious to run our TUI in Concourse. 😂